test/app.js

```diff
@@ -1,103 +1,37 @@
-/* global artifacts contract before beforeEach it assert */
 const { assertRevert } = require('@aragon/test-helpers/assertThrow')
+const { getEventArgument } = require('@aragon/test-helpers/events')
+const deployDAO = require('./helpers/deployDAO')
 
 const CounterApp = artifacts.require('CounterApp.sol')
-const DAOFactory = artifacts.require(
-  '@aragon/core/contracts/factory/DAOFactory'
-)
-const EVMScriptRegistryFactory = artifacts.require(
-  '@aragon/core/contracts/factory/EVMScriptRegistryFactory'
-)
-const ACL = artifacts.require('@aragon/core/contracts/acl/ACL')
-const Kernel = artifacts.require('@aragon/core/contracts/kernel/Kernel')
-
-const getContract = name => artifacts.require(name)
 
 const ANY_ADDRESS = '0xffffffffffffffffffffffffffffffffffffffff'
 
-contract('CounterApp', accounts => {
-  let APP_MANAGER_ROLE, INCREMENT_ROLE, DECREMENT_ROLE
-  let daoFact, appBase, app
-
-  const firstAccount = accounts[0]
-  const secondAccount = accounts[1]
-
-  before(async () => {
-    const kernelBase = await getContract('Kernel').new(true) // petrify immediately
-    const aclBase = await getContract('ACL').new()
-    const regFact = await EVMScriptRegistryFactory.new()
-    daoFact = await DAOFactory.new(
-      kernelBase.address,
-      aclBase.address,
-      regFact.address
-    )
-    appBase = await CounterApp.new()
-
-    // Setup constants
-    APP_MANAGER_ROLE = await kernelBase.APP_MANAGER_ROLE()
-    INCREMENT_ROLE = await appBase.INCREMENT_ROLE()
-    DECREMENT_ROLE = await appBase.DECREMENT_ROLE()
-  })
+contract('CounterApp', ([_, appManager, user]) => {
+  let app
 
-  beforeEach(async () => {
-    const daoReceipt = await daoFact.newDAO(firstAccount)
-    const dao = Kernel.at(
-      daoReceipt.logs.filter(l => l.event === 'DeployDAO')[0].args.dao
-    )
-    const acl = ACL.at(await dao.acl())
+  beforeEach('deploy dao and app', async () => {
+    const { dao, acl } = await deployDAO(appManager)
 
-    await acl.createPermission(
-      firstAccount,
-      dao.address,
-      APP_MANAGER_ROLE,
-      firstAccount,
-      {
-        from: firstAccount,
-      }
-    )
+    // Deploy the app's base contract.
+    const appBase = await CounterApp.new()
 
-    const receipt = await dao.newAppInstance(
-      '0x1234',
-      appBase.address,
-      '0x',
-      false,
-      { from: firstAccount }
-    )
+    // Instantiate a proxy for the app, using the base contract as its logic implementation.
+    const instanceReceipt = await dao.newAppInstance('0x1234', appBase.address, '0x', false, { from: appManager })
```

Comment:
Perhaps we can add some inline documentation on each argument here.

Particularly the `'0x1234'` and `'0x'` might seem arbitrary and confusing for people who don't know how these arguments can be used.

Reply:
Done. I also added documentation to one of the createPermission calls below. Want to take a look @sohkai?

Reply:
Not sure if is the right place but probably good to mention on the tutorial the "convention" of using the `apmNameHash` as the `appId`